song-server/src/main/java/org/icgc/dcc/song/server/controller/AnalysisController.java

```diff
@@ -94,7 +94,8 @@
    * @return A JSON object representing this analysis
    */
   @GetMapping(value = "/{id}")
-  public Analysis read(@PathVariable("id") String id) {
+  @PreAuthorize("@studySecurity.authorize(authentication, #studyId)")
+  public Analysis read(@PathVariable("studyId") String studyId, @PathVariable("id") String id) {
     return analysisService.read(id);
   }
 
```

Comment:
It needs to be there for when reflection occurs and the `@PreAuthorize` annotation references `#studyId` in the spring expression. Otherwise you might get a good old NPE.